src/libraries/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs

```diff
@@ -718,7 +734,45 @@ public virtual unsafe int FindFirstCharacterToEncodeUtf8(ReadOnlySpan<byte> utf8
                         // casted to signed byte.
                         int index = Sse2.MoveMask(sourceValue);
 
-                        if (index != 0)
+                        if (index == 0)
```

Comment:
Flipped the `if`, as this results in better code-gen for the vectorized path (SSSE3).